e2e_playwright/iframe_resizer_test.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import os
+from typing import TYPE_CHECKING, Callable, Final
+
+from playwright.sync_api import FrameLocator, Locator, expect
+
+from e2e_playwright.conftest import IframedPageAttrs
+from e2e_playwright.shared.app_utils import wait_for_app_run
+
+if TYPE_CHECKING:
+    from e2e_playwright.conftest import IframedPage, ImageCompareFunction
+
+TEST_ASSETS_DIR = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), "test_assets"
+)
+
+with open(os.path.join(TEST_ASSETS_DIR, "iframerResizer.min.js")) as f:
+    IFRAME_RESIZER_SCRIPT: Final[str] = f.read()
+
+
+def _open_with_resize_script(
+    iframed_app: IframedPage, embed: bool = False, with_min_height: bool = False
+) -> FrameLocator:
+    """Open the iframe with the resize script and return the frame locator."""
+    src_query_params = {}
+    if embed:
+        src_query_params["embed"] = "true"
+
+    frame_locator: FrameLocator = iframed_app.open_app(
+        IframedPageAttrs(
+            element_id="exampleframe",
+            src_query_params=src_query_params,
+            additional_html_head=f"<script>{IFRAME_RESIZER_SCRIPT}</script>",
+        )
+    )
+    wait_for_app_run(frame_locator)
+
+    resize_args: dict[str, str | int] = {"log": "true"}
+    if with_min_height:
+        resize_args["minHeight"] = 500
+
+    # call the iFrameResize function with the resize arguments
```

Comment:
nit: maybe mention that this method gets exposed to the global scope via the iframerResizer library